crates/binjs_es6/src/scopes.rs

```diff
@@ -1153,81 +1174,39 @@ impl Visitor<ScopeError> for AnnotationVisitor {
         Ok(None)
     }
 
-    fn enter_eager_function_declaration(
+    fn enter_lazy_function_expression(
         &mut self,
         _path: &WalkPath,
-        _node: &mut EagerFunctionDeclaration,
+        node: &mut LazyFunctionExpression,
     ) -> EnterResult {
+        self.push_function_expression_name(node.name.clone());
         Ok(VisitMe::HoldThis(()))
     }
+    fn exit_lazy_function_expression(
+        &mut self,
+        _path: &WalkPath,
+        _node: &mut LazyFunctionExpression,
+    ) -> ExitResult<LazyFunctionExpression> {
+        self.pop_function_expression_name();
+        Ok(None)
+    }
+
     fn exit_eager_function_declaration(
         &mut self,
         path: &WalkPath,
         node: &mut EagerFunctionDeclaration,
     ) -> ExitResult<EagerFunctionDeclaration> {
         debug!(target: "annotating", "exit_eager_function_declaration {:?} at {:?}", node.name.name, path);
+        self.exit_lazy_or_eager_function_declaration(path, &node.name)?;
```

Comment:
In all of these, it seems you could just write:

```rust
self.exit_lazy_or_eager_function_declaration(path, &node.name)
```

and return result immediately?

Reply:
The return type is different, unfortunately.

Reply:
Ah. I guess you can make that function generic to return any `ExitResult<T>`?